javascript/javascript2/week1/spiritAnimalNameGenerator.js

```diff
@@ -0,0 +1,68 @@
+const body = document.querySelector("body");
+
+// create a div to write output to
+const output = document.createElement("div");
+//apply few styles
+output.style.color = "green";
+body.appendChild(output);
+
+function getSpiritAnimalName() {
+  const x = document.getElementById("name").value;
+  const spiritAnimalNames = [
+    "Alligator",
+    "Ant",
+    "Badger",
+    "Bat",
+    "Bear",
+    "Bee",
+    "Blue Jay",
+    "Bobcat",
+    "Buffalo",
+    "Bull",
+    "Butterfly",
+    "Cardinal",
+  ];
+  if (x.length > 0) {
+    output.innerHTML =
+      x + "-" + spiritAnimalNames[Math.ceil(Math.random() * 10)];
+  } else {
+    output.innerHTML = ""; // this is needed to remove previous string when input is clear
+  }
+
+  /* button.disabled = true;
+  const newButton = document.createElement("button");
+  newButton.innerHTML = "Get new spirit animal name for";
+  body.appendChild(newButton);
+  newButton.addEventListener("click", getName); */
+}
+
+// to get the select option value :: to select event
+
+const select = document.getElementById("eventType");
+
+select.addEventListener("change", function () {
+  const button = document.querySelector("button");
+  const textField = document.querySelector("input");
+  if (select.value === "click") {
+    //adding event listener on mouse click
+    button.addEventListener("click", getSpiritAnimalName);
+
+    //remove other two events---hover mouse on button and input on the text field
+    button.removeEventListener("mouseover", getSpiritAnimalName);
+    textField.removeEventListener("input", getSpiritAnimalName);
```

Comment:
I didn't find any difference even if I remove these two lines. removeEventListener is necessary?

Reply:
here the user is given a choice to choose the event from a list of events. if i do not remove above 2, all the events will work irrespective of users choice